### PR TITLE
fix: allow users to specify custom template for the article

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

**Comment by sywhb** [comment]
@jacksonh I'm adding an option set to allow users to create a custom template for the article and it looks the same by default

**Comment by jacksonh** [comment]
Checking this one out now!

**Comment by sywhb** [comment]
Hey @jacksonh, can we merge it?

**Comment by sywhb** [comment]
Hey @jacksonh, do you think we can merge this and test it?

**Comment by jacksonh** [comment]
@sywhb yeah, just have to make sure it wont release when we merge!

**Review by jacksonh — APPROVED** [review]

**Comment by sywhb** [comment]
> @sywhb yeah, just have to make sure it wont release when we merge!

@jacksonh, in this case, I guess we shouldn't merge this now. Please let me know when it is good to go

**Comment by jacksonh** [comment]
Let me try testing it out again locally. Sorry I just saw your updates to the template now. Does this allow for optional handling of items like labels if not null? something like `if labels.length > 0 { write(labels) }`

File: src/index.ts [hunk 1]
```diff
@@ -440,21 +439,23 @@ const main = async (baseInfo: LSPluginBaseInfo) => {
...
       collapsed:: true
-      site:: [{siteName}]({originalUrl})
-      author:: {author}
-      labels:: {labels}
-      date_saved:: {dateSaved}`,
+      site:: [{{{siteName}}}]({{{originalUrl}}})
+      author:: {{{author}}}
+      {{#labels.length}}
+      labels:: {{#labels}}[[{{{name}}}]]{{/labels}}
+      {{/labels.length}}
```
Comment (sywhb) [inline src/index.ts:451]: @jacksonh this code will check the size of labels and prevent showing an empty list of labels on the page

File: src/index.ts [hunk 2]
```diff
@@ -415,6 +429,38 @@ const main = async (baseInfo: LSPluginBaseInfo) => {
...
+      type: 'string',
+      title: 'Enter the template to use for new articles',
+      description:
+        'Enter the template to use for new articles. Required variables are: {{{title}}}, {{{omnivoreUrl}}}. Optional variables are: {{{siteName}}}, {{{originalUrl}}}, {{{author}}}, {{{labels}}}, {{{dateSaved}}}',
+      default: `[{{{title}}}]({{{omnivoreUrl}}})
+      collapsed:: true
+      site:: {{#siteName}}[{{{siteName}}}]{{/siteName}}({{{originalUrl}}})
+      {{#author}}
+      author:: {{{author}}}
+      {{/author}}
```
Comment (sywhb) [inline src/index.ts:450]: @jacksonh Made `siteName` and `author` optional

**Pull request merged**